### PR TITLE
Fix tiling

### LINKED ISSUE #157: Tiling assumes channel first input
See https://github.com/bioimage-io/core-bioimage-io-python/blob/main/bioimageio/core/prediction.py#L222

We need a model that has channel last (stardist example model?!) to write a test and then fix this.

### PR DESCRIPTION
**Pull request by constantinpape**
Fixes #157

### CODE REVIEW

File: tests/test_prediction.py [hunk 1]
```diff
@@ -79,24 +77,34 @@ def check_result():
...
     # test with dynamic padding
-    predict_image(any_model, in_path, out_path, padding={"x": 8, "y": 8, "mode": "dynamic"})
+    predict_image(model, in_path, out_path, padding={"x": 8, "y": 8, "mode": "dynamic"})
     check_result()
 
     # test with fixed padding
     predict_image(
-        any_model, in_path, out_path, padding={"x": original_shape[0], "y": original_shape[1], "mode": "fixed"}
+        model, in_path, out_path, padding={"x": original_shape[0], "y": original_shape[1], "mode": "fixed"}
     )
```
[the remark this answers is not in the document]
Comment (constantinpape) [reply tests/test_prediction.py:104]: something with this suggestion is wrong

**Comment by constantinpape** [comment]
@FynnBe this should be good to go now.
In addition to fixing the tiling for models with channel last this fixes a few more bugs with prediction, in particular enabling prediction with models with multiple input/output tensors, which failed before due to a check done at a wrong position.
It also updates the conftest to make use of the new models and removes FruNet from it, which is deprecated.

**Comment by constantinpape** [comment]
The issues with loading torch models are back .... :(.

**Comment by FynnBe** [comment]
> The issues with loading torch models are back .... :(.

well we are not filtering out pre pytorch 1.10 models here yet...  ( oh darn, just noticed that I called the list pre_3_10 (got confused with the python version I guess^^)
Anyhow, I'm not sure that's really the fix, also because then we just don't test them, now that even python 3.7 goes along with pytorch 1.10...

**Comment by constantinpape** [comment]
> well we are not filtering out pre pytorch 1.10 models here yet... ( oh darn, just noticed that I called the list pre_3_10 (got confused with the python version I guess^^)
> Anyhow, I'm not sure that's really the fix, also because then we just don't test them, now that even python 3.7 goes along with pytorch 1.10...

Maybe we should pin to an older pytorch version in the tests for now.
And we can try to update the relevant models to pytorch 1.10 next week.

**Comment by constantinpape** [comment]
@FynnBe I pinned to "pytorch<3.10" for now so we can run all the tests.
I will try to update the models next week so that we can run with 3.10.
This is good to go from my side now, let me know if you think there are any more changes required.

**Pull request merged**